examples/js/exporters/GLTFExporter.js

```diff
@@ -776,7 +777,35 @@ THREE.GLTFExporter.prototype = {
 
 				}
 
-				ctx.drawImage( image, 0, 0, canvas.width, canvas.height );
+				if ( ( typeof HTMLImageElement !== 'undefined' && image instanceof HTMLImageElement ) ||
+					( typeof HTMLCanvasElement !== 'undefined' && image instanceof HTMLCanvasElement ) ||
+					( typeof ImageBitmap !== 'undefined' && image instanceof ImageBitmap ) ) {
+
+					ctx.drawImage( image, 0, 0, canvas.width, canvas.height );
+
+				} else {
+
+					if ( format !== THREE.RGBAFormat && format !== THREE.RGBFormat )
+						throw "Only RGB and RGBA formats are supported";
+
+					if ( image.width !== canvas.width || image.height !== canvas.height )
+						console.warn( "Image size and imposed canvas sized do not match" );
```

Comment:
One more question. When could this happen?

Reply:
When the option `maxTextureSize` is used.

Reply:
I see... 

This would be more clear, right?

```js
if ( image.width > options.maxTextureSize || image.height > options.maxTextureSize ) {

	console.warn( 'GLTFExporter: Image size is bigger than maxTextureSize', image );

}
```

Reply:
Indeed 👍 .